pkg/errors/sdk_errors.go

```diff
@@ -80,15 +81,15 @@ func CheckError(resp *http.Response, expectedStatusCodes ...int) SDKError {
 
 	var content map[string]interface{}
 	if err := json.NewDecoder(resp.Body).Decode(&content); err != nil {
-		return NewSDKErrorWithStatus(err.Error(), resp.StatusCode)
+		return NewSDKErrorWithStatus(err, resp.StatusCode)
 	}
 
 	if msg, ok := content[err]; ok {
 		if v, ok := msg.(string); ok {
-			return NewSDKErrorWithStatus(v, resp.StatusCode)
+			return NewSDKErrorWithStatus(errors.New(v), resp.StatusCode)
 		}
-		return NewSDKErrorWithStatus(unknownErr, resp.StatusCode)
+		return NewSDKErrorWithStatus(errors.New(unknownErr), resp.StatusCode)
```

Comment:
Since `NewSDKError` and `NewSDKErrorWithStatus` now accept error - `unknownErr` and `jsonKeyNotFound` should be errors again.